File: src/main/java/jpabook/start/pr4/Team.java
package jpabook.start.pr4;

import java.util.List;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.Id;
import javax.persistence.OneToMany;
import javax.persistence.Table;

/**
 * @author 최영서 (youngseo.choi@navercorp.com)
 */
@Entity
@Table(name = "team")
public class Team {
    @Id
    @Column(name = "team_id")
    private String id;

    private String name;

    @OneToMany(mappedBy = "team", fetch = FetchType.LAZY)
    private List<Member> members;

    public String finalGetName() {
        return name;
    }

    public void finalSetName(String name) {
        this.name = name;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public List<Member> getMembers() {
        return members;
    }

    public void setMembers(List<Member> members) {
        this.members = members;
    }
}

File: src/main/kotlin/jpabook/start/pr4/Team.kt
package jpabook.start.pr4

import org.hibernate.annotations.AttributeAccessor
import javax.persistence.*

@Entity
@Table(name = "team")
//@Access(AccessType.FIELD)
class Team(
    @Id
    @Column(name = "team_id")
    val id: String,

    val name: String,

    @OneToMany(mappedBy = "team", fetch = FetchType.LAZY)
    val members: MutableList<Member> = mutableListOf()
)